core/src/banking_stage/consumer.rs

```diff
@@ -1065,6 +1092,164 @@ mod tests {
+                                panic!("Error returning mixin hash");
+                            }
+                        }
+                        if is_exited.load(Ordering::Relaxed) {
+                            break;
+                        }
+                    });
+                tick_producer.unwrap()
+            }
+
+            // Simulate a race condition by setting up poh to do the last tick
+            // right before returning the transaction record response so that
+            // bank blockhash queue is updated before transactions are
+            // committed.
+            let poh_simulator =
+                poh_tick_before_returning_record_response(record_receiver, &poh_recorder);
+
+            poh_recorder
+                .write()
+                .unwrap()
+                .set_bank_for_test(bank.clone());
+
+            // Tick up to max tick height - 1 so that only one tick remains
+            // before recording transactions to poh
+            while poh_recorder.read().unwrap().tick_height() != bank.max_tick_height() - 1 {
+                poh_recorder.write().unwrap().tick();
+            }
+
+            let (replay_vote_sender, _replay_vote_receiver) = unbounded();
+            let committer = Committer::new(
+                None,
+                replay_vote_sender,
+                Arc::new(PrioritizationFeeCache::new(0u64)),
+            );
+            let consumer = Consumer::new(committer, recorder, QosService::new(1), None);
+
+            let process_transactions_batch_output =
+                consumer.process_and_record_transactions(&bank, &transactions, 0);
+            let ExecuteAndCommitTransactionsOutput {
+                transactions_attempted_execution_count,
+                executed_transactions_count,
+                executed_with_successful_result_count,
+                commit_transactions_result,
+                ..
+            } = process_transactions_batch_output.execute_and_commit_transactions_output;
+
+            assert_eq!(transactions_attempted_execution_count, 1);
+            assert_eq!(executed_transactions_count, 1);
+            assert_eq!(executed_with_successful_result_count, 0);
+            assert!(commit_transactions_result.is_ok());
+
+            // Ensure that poh did the last tick after recording transactions
+            assert_eq!(
+                poh_recorder.read().unwrap().tick_height(),
+                bank.max_tick_height()
+            );
+
+            let mut done = false;
+            // read entries until I find mine, might be ticks...
+            while let Ok((_bank, (entry, _tick_height))) = entry_receiver.recv() {
+                if !entry.is_tick() {
+                    assert_eq!(entry.transactions.len(), transactions.len());
+                    done = true;
+                }
+                if done {
+                    break;
+                }
+            }
+            assert!(done);
```

Comment:
*bikeshedding*
```suggestion
            let my_enrty = entry_receiver
                .iter()
                .map(|(_bank, (entry, _tick_height))| entry)
                .filter(|entry| !entry.is_tick())
                .first()
                .expect("A non-tick entry was inserted [where?]");
            assert_eq!(my_entry.transactions.len(), transactions.len());
```

Maybe there is a better name than `my_entry`?  But I do not fully understand the code to suggset it.

Reply:
The code looks through the entries of a block to find one that has transactions in it.. since the test only sends one transaction, it's enough to know that the transaction was included. It's not really necessary to have this check because there are other checks for whether the tx was committed but it's here because I copied it from another test. It's not really important though, I think it's fine to leave as is.

Reply:
Yeah, it is fine to leave it as is.
Though, I think, the version I wrote is functionally identical to the one in the PR.
If by "this check" you mean the `.expect("A non-tick entry was inserted [where?]");` - existing version also has this check.  It is the `assert!(done)` at the end.
I just thought, an explicit explanation is almost always better.